java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,602 @@
+        TryStatementTree tryStatementTree = (TryStatementTree) tree;
+        currentBlock = createBlock(currentBlock);
+        BlockTree finallyBlock = tryStatementTree.finallyBlock();
+        if(finallyBlock != null) {
+          build(finallyBlock);
+        }
+        currentBlock = createBlock(currentBlock);
+        build(tryStatementTree.block());
+        break;
+      }
+      case THROW_STATEMENT: {
+        //FIXME this won't work if it is intended to be caught by a try statement.
+        ThrowStatementTree throwStatementTree = (ThrowStatementTree) tree;
+        currentBlock = createUnconditionalJump(throwStatementTree, exitBlock);
+        build(throwStatementTree.expression());
+        break;
+      }
+      case SYNCHRONIZED_STATEMENT: {
+        SynchronizedStatementTree sst = (SynchronizedStatementTree) tree;
+        //Naively build synchronized statement.
+        build(sst.block());
+        build(sst.expression());
+        break;
+      }
+      case POSTFIX_INCREMENT:
+      case POSTFIX_DECREMENT:
+      case PREFIX_INCREMENT:
+      case PREFIX_DECREMENT:
+      case UNARY_MINUS:
+      case UNARY_PLUS:
+      case BITWISE_COMPLEMENT:
+      case LOGICAL_COMPLEMENT:
+        UnaryExpressionTree e = (UnaryExpressionTree) tree;
+        currentBlock.elements.add(e);
+        build(e.expression());
+        break;
+      case PARENTHESIZED_EXPRESSION:
+        build(((ParenthesizedTree) tree).expression());
+        break;
+      case ARRAY_ACCESS_EXPRESSION: {
+        ArrayAccessExpressionTree aaet = (ArrayAccessExpressionTree) tree;
+        currentBlock.elements.add(aaet);
+        build(aaet.expression());
+        build(aaet.dimension());
+        break;
+      }
+      case ARRAY_DIMENSION: {
+        ArrayDimensionTree arrayDimensionTree = (ArrayDimensionTree) tree;
+        build(arrayDimensionTree.expression());
+        break;
+      }
+      case IDENTIFIER:
+      case INT_LITERAL:
+      case LONG_LITERAL:
+      case DOUBLE_LITERAL:
+      case CHAR_LITERAL:
+      case FLOAT_LITERAL:
+      case STRING_LITERAL:
+      case BOOLEAN_LITERAL:
+      case NULL_LITERAL:
+        currentBlock.elements.add(tree);
+        break;
+      default:
+        currentBlock.elements.add(tree);
+        break;
+        //throw new UnsupportedOperationException(tree.kind().name());
+    }
+
+  }
+
+  private Block createUnconditionalJump(Tree terminator, @Nullable Block target) {
+    Block result = createBlock();
+    result.terminator = terminator;
+    if (target != null) {
+      result.successors.add(target);
+    }
+    return result;
+  }
+
+  private void buildCondition(Tree syntaxNode, Block trueBlock, Block falseBlock) {
+    switch (syntaxNode.kind()) {
+      case CONDITIONAL_OR: {
+        BinaryExpressionTree e = (BinaryExpressionTree) syntaxNode;
+        // process RHS
+        buildCondition(e.rightOperand(), trueBlock, falseBlock);
+        falseBlock = currentBlock;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "falseBlock". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)